### PR TITLE
SecurityInsights: Investigation swagger updates

### PR DESCRIPTION
**Pull request by dosegal**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by nschonni** [comment]
You can add the extra words to https://github.com/Azure/azure-rest-api-specs/blob/master/custom-words.txt if they're correct

**Comment by dosegal** [comment]
@shahabhijeet please review, thanks :)

**Comment by dosegal** [comment]
@shahabhijeet is there any update?

**Review by majastrz — APPROVED** [review]
Looks good from ARM side.

**Comment by dosegal** [comment]
@shahabhijeet @praries880 can someone please review the PR? thanks

**Review by mmyyrroonn — APPROVED** [review]
LGTM. I just want to confirm. Lot's of parameters set `readOnly` field to `true`. Usually, `readOnly: true` means these parameters should not appear in the PUT/PATCH methods. These parameters are set by design. Right?

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 1]
```diff
@@ -2678,34 +2769,122 @@
...
+    "EntityCommonProperties": {
+      "type": "object",
+      "description": "Entity common property bag.",
+      "properties": {
+        "friendlyName": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The graph item display name which is a short humanly readable description of the graph item instance. This property is optional and might be system generated."
+        },
+        "additionalData": {
```
[the remark this answers is not in the document]
Comment (dosegal) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: additionalData can hold any custom field that should be part of the "entity" object. The fields don't share the same type, so this is why the type of "additionalProperties" is object.

**Review by shahabhijeet — APPROVED** [review]

**Comment by shahabhijeet** [comment]
@MyronFanQiu go ahead merge the PR if you are satisfied with responses to your comments.

**Comment by dosegal** [comment]
> LGTM. I just want to confirm. Lot's of parameters set `readOnly` field to `true`. Usually, `readOnly: true` means these parameters should not appear in the PUT/PATCH methods. These parameters are set by design. Right?

@MyronFanQiu One of our resource types, called `entities`, supports only Get/Post operations in the server-side, and creation/modification does not allowed as part of the API. We have 15 different kinds of this resource, and they all share the same behavior. This is why you see many `readOnly` properties set to true.

**Comment by MyronFanQiu** [comment]
@dosegal  Thanks!

**Pull request merged**